addons/mail/models/mail_template.py

```diff
@@ -89,6 +89,34 @@ def default_get(self, fields):
     can_write = fields.Boolean(compute='_compute_can_write',
                                help='The current user can edit the template.')
 
+    def _constrain_template_rendering_generate_template(self):
+        self.ensure_one()
+        model = self.model_id.model
+        if not model:
+            return
+        record = self.env[model].search([], limit=1)
+        if not record.ids:
+            return
+        mail_template_fields = [
+            'body_html',
+            'subject',
+            'email_cc',
+            'email_from',
+            'email_to',
+            'partner_to'
+        ]
+        return self._generate_template(record.ids, mail_template_fields)
```

Comment:
Generate_template create side records, notably partners. We don't want that, so you should probably have a rollback mechanism to avoid altering DB.

Reply:
@tde-banana-odoo  done